include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -2165,6 +2180,77 @@ namespace VectorTools
 
   namespace
   {
+    // keep first argument is non-reference since we use it
+    // with 1e-8 * number
+    template <typename number1, typename number2>
+    bool real_part_bigger_than(const number1 a,
+                               const number2 &b)
+    {
+      return a > b;
+    }
+
+    template <typename number1, typename number2>
+    bool real_part_bigger_than(const std::complex<number1> a,
+                               const std::complex<number2> &b)
+    {
+      Assert(std::abs(a.imag()) < 1e-15 , ExcInternalError());
+      Assert(std::abs(b.imag()) < 1e-15 , ExcInternalError());
+      return a.real() > b.real();
+    }
+
+    template <typename number>
+    double min_number()
+    {
+      return std::numeric_limits<number>::min();
```

Comment:
also, what do you do if you can express the minimal value of a type in `double`? E.g., if you have a long double? In that case, the return value is converted to `double` precision, which makes it zero.